core/mondoo-ms365-security.mql.yaml

```diff
@@ -0,0 +1,625 @@
+          ```
+        __cnquery shell__
+
+        To audit Microsoft ms365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          msgraph.beta.devicemanagement.deviceConfigurations.where( properties['@odata.type'] == "#microsoft.graph.androidGeneralDeviceConfiguration") { properties['storageRequireDeviceEncryption'] == true }
+          ```
+      remediation: |
+        ### MS365 Console
+         
+         To update via the ms365 portal:
+
+         1. Log in as to the ms365 portal at https://admin.microsoft.com
+         2. Endpoint Manager --> Devices --> Policy --> Configuration profiles 
+         3. Ensure that a profile exists for Android with following conditions:
+            * Password section --> Device restrictions --> Encryption is set to require
+    query: |
+      msgraph.beta.devicemanagement.deviceConfigurations.where( properties['@odata.type'] == "#microsoft.graph.androidGeneralDeviceConfiguration") {
+        properties['storageRequireDeviceEncryption'] == true
+      }
+  - uid: mondoo-m365-ensure-third-party-integrated-applications-are-not-allowed
+    title: Ensure that no third party integrated applications are allowed 
+    severity: 80
+    docs:
+      desc: |
+        This control checks that no third party integrated applications can connect to your services. 
+      audit: |
+        __cnquery run__
+
+        To audit Microsoft MS365 with `cnquery run`:
+
+        Run the following query:
+
+          ```mql
+          cnquery run ms365 -c "msgraph.beta.policies.authorizationPolicy { _['defaultUserRolePermissions']['allowedToCreateApps'] == false }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        __cnquery shell__
+
+        To audit Microsoft ms365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          msgraph.beta.policies.authorizationPolicy { _['defaultUserRolePermissions']['allowedToCreateApps'] == false }
+          ```
+      remediation: |
+        ### MS365 Console
+
+        To update via the ms365 portal:
+
+        1. Log in as Global Administrator to the ms365 portal at https://admin.microsoft.com
+        2. Click on "Azure Active Directory" once in the "Admin Centers"
+        3. Users --> User settings -->  App registrations set to NO
+        4. Save
+    query: |
+      msgraph.beta.policies.authorizationPolicy { 
```

Comment:
` msgraph.beta.policies` is deprecated, use latest version

Reply:
done!